### PR TITLE
Add `@AllowUnusedBindings` sqlobject customizer

### PR DESCRIPTION
**Pull request by stevenschlansker**
From #1418 

@sarxos, hopefully you will find this useful?

### CODE REVIEW

File: sqlobject/src/test/java/org/jdbi/v3/sqlobject/customizer/TestAllowUnusedBindings.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    @Test
+    public void testAllowed() {
+        assertThat(dao.allowed("42")).isTrue();
+    }
+
+    @Test
+    public void testDisallowed() {
+        assertThatThrownBy(() -> dao.disallowed("43"))
```
Comment (TheRealMarnes) [inline sqlobject/src/test/java/org/jdbi/v3/sqlobject/customizer/TestAllowUnusedBindings.java]: I'd set it to `true` explicitly on the handle instance to really prove that the annotation overrides it

Comment (TheRealMarnes) [reply sqlobject/src/test/java/org/jdbi/v3/sqlobject/customizer/TestAllowUnusedBindings.java]: Setting it on the handle _after_ the dao instance has been created is a moot action, isn't it?

Comment (stevenschlansker) [reply sqlobject/src/test/java/org/jdbi/v3/sqlobject/customizer/TestAllowUnusedBindings.java]: made it `onDemand` instead

**Comment by stevenschlansker** [comment]
The Travis failures seem unrelated to this PR. :(

**Comment by sarxos** [comment]
Thank you @stevenschlansker!

**Pull request merged**